operator/controllers/runtimetask_controller_test.go

```diff
@@ -0,0 +1,820 @@
+	}
+	type want struct {
+		task *operatorv1.RuntimeTask
+	}
+	tests := []struct {
+		name string
+		args args
+		want want
+	}{
+		{
+			name: "Reconcile pending state",
+			args: args{
+				task: &operatorv1.RuntimeTask{
+					Status: operatorv1.RuntimeTaskStatus{},
+				},
+			},
+			want: want{
+				task: &operatorv1.RuntimeTask{
+					Status: operatorv1.RuntimeTaskStatus{
+						Phase: string(operatorv1.RuntimeTaskPhasePending),
+					},
+				},
+			},
+		},
+		{
+			name: "Reconcile running state",
+			args: args{
+				task: &operatorv1.RuntimeTask{
+					Status: operatorv1.RuntimeTaskStatus{
+						StartTime: tx,
+					},
+				},
+			},
+			want: want{
+				task: &operatorv1.RuntimeTask{
+					Status: operatorv1.RuntimeTaskStatus{
+						StartTime: tx,
+						Phase:     string(operatorv1.RuntimeTaskPhaseRunning),
+					},
+				},
+			},
+		},
+		{
+			name: "Reconcile paused state",
+			args: args{
+				task: &operatorv1.RuntimeTask{
+					Status: operatorv1.RuntimeTaskStatus{
+						StartTime: tx,
+						Paused:    true,
+					},
+				},
+			},
+			want: want{
+				task: &operatorv1.RuntimeTask{
+					Status: operatorv1.RuntimeTaskStatus{
+						StartTime: tx,
+						Paused:    true,
+						Phase:     string(operatorv1.RuntimeTaskPhasePaused),
+					},
+				},
+			},
+		},
+		{
+			name: "Reconcile succeeded state",
+			args: args{
+				task: &operatorv1.RuntimeTask{
+					Status: operatorv1.RuntimeTaskStatus{
+						StartTime:      tx,
+						CompletionTime: tx,
+					},
+				},
+			},
+			want: want{
+				task: &operatorv1.RuntimeTask{
+					Status: operatorv1.RuntimeTaskStatus{
+						StartTime:      tx,
+						CompletionTime: tx,
+						Phase:          string(operatorv1.RuntimeTaskPhaseSucceeded),
+					},
+				},
+			},
+		},
+		{
+			name: "Reconcile failed state",
+			args: args{
+				task: &operatorv1.RuntimeTask{
+					Status: operatorv1.RuntimeTaskStatus{
+						StartTime:    tx,
+						ErrorMessage: stringPtr("error"),
+					},
+				},
+			},
+			want: want{
+				task: &operatorv1.RuntimeTask{
+					Status: operatorv1.RuntimeTaskStatus{
+						StartTime:    tx,
+						ErrorMessage: stringPtr("error"),
+						Phase:        string(operatorv1.RuntimeTaskPhaseFailed),
+					},
+				},
+			},
+		},
+		{
+			name: "Reconcile deleted state",
+			args: args{
+				task: &operatorv1.RuntimeTask{
+					ObjectMeta: metav1.ObjectMeta{
+						DeletionTimestamp: tx,
+					},
+				},
+			},
+			want: want{
+				task: &operatorv1.RuntimeTask{
+					ObjectMeta: metav1.ObjectMeta{
+						DeletionTimestamp: tx,
+					},
+					Status: operatorv1.RuntimeTaskStatus{
+						Phase: string(operatorv1.RuntimeTaskPhaseDeleted),
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := &RuntimeTaskReconciler{}
+			r.reconcilePhase(tt.args.task)
+
+			if !reflect.DeepEqual(tt.args.task, tt.want.task) {
+				t.Errorf("reconcilePhase() = %v, want %v", tt.args.task, tt.want.task)
```

Comment:
we should follow this pattern:
https://github.com/kubernetes/kubeadm/pull/1923/files#diff-1175c7d3cbb2ee3af06c7f1bbd008d60R134

including the function name is not needed.

Reply:
I don't see any problem in having the name of the function in the error message; actually I think that having the function name can be useful in detecting and fixing the problem